src/traces/surface/convert.js

```diff
@@ -249,10 +245,30 @@ proto.update = function(data) {
         dynamicColor: [[1, 1, 1, 1], [1, 1, 1, 1], [1, 1, 1, 1]],
         dynamicWidth: [1, 1, 1],
         dynamicTint: [1, 1, 1],
-        opacity: 1,
-        colorBounds: [data.zmin * scaleFactor[2], data.zmax * scaleFactor[2]]
+        opacity: 1
+        // TODO: Need to think about how to calculate this
+        /*
+        intensityBounds: [
+          data.zmin * scaleFactor[2],
+          data.zmax * scaleFactor[2]]
+        */
     };
 
+    //Refine if necessary
+    if('intensity' in data) {
+        var intensity = ndarray(
+          new Float32Array(xlen * ylen), [xlen, ylen]);
+        fill(intensity, function(row, col) {
+            return data.intensity[col][row];
+        });
+        coords.push(intensity);
+    }
```

Comment:
In order to preserve backward compatibility, we'll need to set the `intensity` field to `data.z` when `data.surfacecolor` is not set, before passing it to `gl-surface3d`.